packages/components/src/checkbox/src/Checkbox.tsx

```diff
@@ -0,0 +1,128 @@
+import { forwardRef, type ForwardedRef } from "react";
+import { type StyledComponentProps, useStyledSystem, type ResponsiveProp, useResponsiveValue } from "@hopper-ui/styled-system";
+import { CheckboxContext } from "./CheckboxContext.ts";
+import { useContextProps, Checkbox as RACCheckbox, type CheckboxProps as RACCheckboxProps, composeRenderProps } from "react-aria-components";
+import { cssModule } from "../../utils/src/cssModule.ts";
+import { composeClassnameRenderProps, SlotProvider } from "../../utils/index.ts";
+import { Text, TextContext } from "../../Text/index.ts";
+import { isTextOnlyChildren } from "../../utils/src/isTextOnlyChildren.ts";
+import { IconContext, CheckmarkIcon, MinusIcon } from "@hopper-ui/icons";
+import { IconListContext } from "../../IconList/index.ts";
+
+import styles from "./Checkbox.module.css";
+
+export const GlobalCheckboxCssSelector = "hop-Checkbox";
+
+// Won't be needed in next react-aria-components release: https://github.com/adobe/react-spectrum/pull/5850
+const DefaultCheckboxSlot = "checkbox";
+
+export interface CheckboxProps extends StyledComponentProps<RACCheckboxProps> {
+    /**
+     * A checkbox can vary in size.
+     */
+    size?: ResponsiveProp<"sm" | "md">;
+}
+
+function Checkbox(props:CheckboxProps, ref: ForwardedRef<HTMLLabelElement>) {
+    [props, ref] = useContextProps({ ...props, slot: props.slot || DefaultCheckboxSlot }, ref, CheckboxContext);
+    const { stylingProps, ...ownProps } = useStyledSystem(props);
+    const {
+        className,
+        children: childrenProp,
+        size: sizeProp = "md",
+        style: styleProp,
+        ...otherProps
+    } = ownProps;
+
+    
+    const size = useResponsiveValue(sizeProp) ?? "md";
+
+    const classNames = composeClassnameRenderProps(
+        className,
+        GlobalCheckboxCssSelector,
+        cssModule(
+            styles,
+            "hop-Checkbox",
+            size
+        ),
+        stylingProps.className
+    );
+
+    const style = composeRenderProps(styleProp, prev => {
+        return {
+            ...stylingProps.style,
+            ...prev
+        };
+    });
+
+    const children = composeRenderProps(childrenProp, prev => {
+        if (isTextOnlyChildren(prev)) {
+            return <Text>{prev}</Text>;
+        }
+
+        return prev;
+    });
+
+    return (
+        
+        <RACCheckbox
+            ref={ref}
+            className={classNames}
+            style={style}
+            {...otherProps}
+        >
+            {checkboxProps => {
+                const checkboxIconClassName = styles["hop-Checkbox__check"];
+                const icon = checkboxProps.isIndeterminate ? 
+                    <MinusIcon size="sm" className={checkboxIconClassName} /> : 
+                    <CheckmarkIcon size="sm" className={checkboxIconClassName} />;
+
+                return (
+                    <>
+                        <div className={styles["hop-Checkbox__box"]}>{icon}</div>
```

Comment:
nit pick, but is there a reason for the wrapping div? can't we just put the icon directly?

Reply:
The fix is for the actual box style so border and background. I would not want to style the icon as the box, especially since in medium size, the  box is larger than the icon.

Reply:
NVM 🤦